apache-rat-core/src/main/java/org/apache/rat/report/ConfigurationReport.java

```diff
@@ -0,0 +1,285 @@
+                }
+                writer.closeElement(); // APPROVED
+
+                // write matchers section
+                MatcherBuilderTracker tracker = MatcherBuilderTracker.INSTANCE;
+                writer.openElement(XMLConfigurationReader.MATCHERS);
+                for (Class<?> clazz : tracker.getClasses()) {
+                    writer.openElement(XMLConfigurationReader.MATCHER)
+                            .attribute(XMLConfigurationReader.ATT_CLASS_NAME, clazz.getCanonicalName()).closeElement();
+                }
+                writer.closeElement(); // MATCHERS
+
+                writer.closeElement(); // ROOT
+            } catch (IOException e) {
+                throw new RatException(e);
+            }
+        }
+    }
+
+    private void writeFamily(ILicenseFamily family) throws RatException {
+        try {
+            writer.openElement(XMLConfigurationReader.FAMILY)
+                    .attribute(XMLConfigurationReader.ATT_ID, family.getFamilyCategory().trim())
+                    .attribute(XMLConfigurationReader.ATT_NAME, family.getFamilyName());
+            writer.closeElement();
+        } catch (IOException e) {
+            throw new RatException(e);
+        }
+    }
+
+    private void writeDescriptions(Collection<Description> descriptions, Component component) throws RatException {
+        for (Description description : descriptions) {
+            writeDescription(description, component);
+        }
+    }
+
+    private void writeChildren(Description description, Component component) throws RatException {
+        writeDescriptions(description.childrenOfType(Type.Parameter), component);
+        writeDescriptions(description.childrenOfType(Type.Unlabeled), component);
+        writeDescriptions(description.childrenOfType(Type.Matcher), component);
+        writeDescriptions(description.childrenOfType(Type.License), component);
+    }
+
+    private void writeComment(Description description) throws IOException {
+        if (StringUtils.isNotBlank(description.getDescription())) {
+            writer.comment(description.getDescription());
+        }
+    }
+
+    private void writeContent(Description description, Component component) throws IOException {
+        String paramValue = description.getParamValue(configuration.getLog(), component);
+        if (paramValue != null) {
+            writer.content(paramValue);
+        }
+    }
+
+    private void writeAttribute(Description description, Component component) throws IOException {
+        String paramValue = description.getParamValue(configuration.getLog(), component);
+        if (paramValue != null) {
+            writer.attribute(description.getCommonName(), paramValue);
+        }
+    }
+
+    /* package privete for testing */
+    void writeDescription(Description description, Component component) throws RatException {
+        try {
+            switch (description.getType()) {
+            case Matcher:
```

Comment:
Reading the usage of these enums I'd prefer to use Uppercase values to not mix it with class names (or confuse readers of the sources ;))